Type exports are broken and incorrect in v3

When using a bundler, one is supposed to use:

```js
import "urlpattern-polyfill";
```

But the types are not loaded this way because TypeScript will try to load `urlpattern-polyfill/index.d.ts`:

```
Cannot find module 'urlpattern-polyfill' or its corresponding type declarations.ts
```

This works correctly:

```js
import "urlpattern-polyfill/dist";
```

But then it fails in Node because of the `exports` key in package.json.

---

In short I think you need to set the `types` property in package.json

Additionally, currently there's no way to use the underlying implementation directly without applying a polyfill because:

- there's no `index.impl.d.ts`
- the `index.impl.js` file does not appear in the `exports` map

--- a/package.json
+++ b/package.json
@@ -7,12 +7,12 @@
     "url": "https://github.com/kenchris/urlpattern-polyfill"
   },
   "type": "module",
+  "main": "./dist/index.cjs",
+  "module": "./dist/index.js",
+  "types": "./dist/index.d.ts",
   "exports": {
     "import": "./dist/index.js",
-    "require": "./dist/index.cjs",
-    "main": "./dist/index.cjs",
-    "module": "./dist/index.js",
-    "types": "./dist/index.d.ts"
+    "require": "./dist/index.cjs"
   },
   "files": [
     "dist"

